etna/core/models/location.py

```diff
@@ -0,0 +1,209 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+from wagtail import blocks
+from wagtail.admin.panels import FieldPanel, MultiFieldPanel, TitleFieldPanel
+from wagtail.api import APIField
+from wagtail.api.v2.serializers import StreamField as StreamFieldSerializer
+from wagtail.fields import StreamField
+from wagtail.snippets.models import register_snippet
+
+from etna.core.blocks import (
+    SimplifiedAccordionBlock,
+)
+from etna.core.serializers import ImageSerializer
+
+
+@register_snippet
+class Location(models.Model):
+    """
+    A model representing a location (physical or online).
+    """
+
+    space_name = models.CharField(max_length=255, verbose_name=_("Space name"))
+
+    at_tna = models.BooleanField(
+        default=False,
+        verbose_name=_("at The National Archives"),
+        help_text=_("Check if this venue is at The National Archives."),
+    )
+
+    online = models.BooleanField(
+        default=False,
+        verbose_name=_("online venue"),
+        help_text=_("Check if this is an online venue."),
+    )
+
+    # Address fields
+    first_line = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        verbose_name=_("first line of address"),
+        help_text=_("First line of address for the location, if applicable."),
+    )
+
+    second_line = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        verbose_name=_("second line of address"),
+        help_text=_("Second line of address for the location, if applicable."),
+    )
+
+    postcode = models.CharField(
+        max_length=20,
+        blank=True,
+        null=True,
+        verbose_name=_("postcode"),
+        help_text=_("Postcode for the location, if applicable."),
+    )
+
+    online_detail = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        verbose_name=_("online detail"),
+        help_text=_("Additional detail for online venues, if applicable."),
+    )
+
+    image = models.ForeignKey(
+        "wagtailimages.Image",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        verbose_name=_("Image"),
+        help_text=_("An image representing the location, if applicable."),
+    )
```

Comment:
Is this the only way of achieving this? It's producing a really undesirable and long `<select>` list in the admin.

Reply:
Yeah, this will be the "Plan your visit" section that we have on Exhibition pages - this change will also need to be made there, not sure if you want to do that here too?